### PR TITLE
Exit ongoing for loop on state change

### LINKED ISSUE #577: Break loops if state changes
# Rationale

There might be cases where a loop or computation goes on for too long, and the state changes. In this case, we need to break execution of that operation.

# Implementation

- [ ] Identify and list all the loops that fall under this category of operations
- [ ] label all these loops
- [ ] start a timer based on the block number vs state (progress) and calculate the time remaining for state to get over
- [ ] when the time remaining is < 0, break the loop

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

Added a timer to `for` loop which takes enough time to execute and passes the current state time.

Following are the functions where timer is implemented
- [x] propose - GetIteration
- [ ] Dispute
- [x] GetBiggestStakeAndId
- [ ]  GetSortedVotes
- [x] GetLastProposedEpoch

Fixes #577

### CODE REVIEW

**Comment by ashish10677** [comment]
@Yashk767 please resolve the conflicts

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: cmd/propose.go [hunk 1]
```diff
@@ -54,14 +56,18 @@ func (*UtilsStruct) Propose(client *ethclient.Client, config types.Configuration
 
 	log.Debugf("Biggest staker Id: %d Biggest stake: %s, Stake: %s, Staker Id: %d, Number of Stakers: %d, Salt: %s", biggestStakerId, biggestStake, staker.Stake, staker.Id, numStakers, hex.EncodeToString(salt[:]))
 
+	bufferPercent, err := cmdUtils.GetBufferPercent()
```
[the remark this answers is not in the document]
Comment (Yashk767) [reply cmd/propose.go:59]: We need both buffer percent and state buffer to calculate the remaining time of state. 
Used both in state remaining time calculation in ` GetRemainingTimeOfCurrentState()` in common.go.

**Comment by SkandaBhat** [comment]
1. For each loop, add a blocking wait call
2. Run staker normally
3. test if each loop breaks 
4. test functionality of staker

**Comment by Yashk767** [comment]
> 1. For each loop, add a blocking wait call
> 2. Run staker normally
> 3. test if each loop breaks
> 4. test functionality of staker

It is working fine.

**Pull request merged**